### PR TITLE
update vendor deps

### PR DESCRIPTION
**Pull request by xh3b4sd**
Towards https://github.com/giantswarm/giantswarm/issues/7963.

### CODE REVIEW

**Comment by xh3b4sd** [comment]
I would need some help here. Can't get `dep` do the do. @sslavic you might know how to get it sorted?

File: service/controller/drainer.go [hunk 1]
```diff
@@ -82,10 +61,12 @@ func NewDrainer(config DrainerConfig) (*Drainer, error) {
 	var operatorkitController *controller.Controller
 	{
 		c := controller.Config{
-			Informer:     newInformer,
+			K8sClient:    config.K8sClient,
 			Logger:       config.Logger,
 			ResourceSets: resourceSets,
-			RESTClient:   config.K8sClient.K8sClient().CoreV1().RESTClient(),
+			NewRuntimeObjectFunc: func() runtime.Object {
+				return new(corev1.Pod)
```
Comment (tomahawk28) [inline service/controller/drainer.go]: is it Pod? I thought it should be `KVMConfigs` in here.

**Comment by sslavic** [comment]
> I would need some help here. Can't get `dep` do the do. @sslavic you might know how to get it sorted?

I'm on it ...

File: service/controller/drainer.go (hunk 1 above)
Comment (xh3b4sd) [reply service/controller/drainer.go]: The drainer controller on KVM watches pods. You can check the resources being configured for the resource set.

File: service/controller/cluster.go [hunk 2]
```diff
@@ -358,7 +339,9 @@ func NewCluster(config ClusterConfig) (*Cluster, error) {
 				resourceSetV25,
 				resourceSetV26,
 			},
-			RESTClient: config.K8sClient.G8sClient().ProviderV1alpha1().RESTClient(),
+			NewRuntimeObjectFunc: func() runtime.Object {
+				return new(v1alpha1.KVMConfig)
```
Comment (tomahawk28) [inline service/controller/cluster.go]: I was confused this one. sorry.

File: service/controller/deleter.go [hunk 3]
```diff
@@ -49,29 +48,8 @@ func NewDeleter(config DeleterConfig) (*Deleter, error) {
 		return nil, microerror.Maskf(invalidConfigError, "%T.K8sClient must not be empty", config)
 	}
 
-	if config.ProjectName == "" {
-		return nil, microerror.Maskf(invalidConfigError, "%T.ProjectName must not be empty", config)
-	}
```
Comment (kopiczko) [inline service/controller/deleter.go]: Why is this removed? Checked in a parent package?

**Review by kopiczko — APPROVED** [review]

File: service/controller/deleter.go (hunk 3 above)
Comment (xh3b4sd) [reply service/controller/deleter.go]: I remove these everywhere I find it. This is configured to the operatorkit controller and this is checking if the name is given or not. No reason to do it all the way through all the layers.

Comment (xh3b4sd) [reply service/controller/deleter.go]: I remove these everywhere I find it. This is configured to the operatorkit controller and this is checking if the name is given or not. No reason to do it all the way through all the layers.

Comment (kopiczko) [reply service/controller/deleter.go]: This is still confusing to me.

Comment (xh3b4sd) [reply service/controller/deleter.go]: We will soon use the `pkg/project` approach anyway. Don't waste your energy on this.

Comment (kopiczko) [reply service/controller/deleter.go]: I already approved the PR. I'm talking about general approach.

**Comment by sslavic** [comment]
@xh3b4sd dependency issues have been resolved

**Comment by sslavic** [comment]
Tested on geckon

```
{⎈ |giantswarm-geckon:default} ➜  kvm-operator git:(vendor) k describe pod kvm-operator-685c8f7658-kdkkl -n giantswarm
...
Events:
  Type    Reason     Age   From                 Message
  ----    ------     ----  ----                 -------
  Normal  Scheduled  38m   default-scheduler    Successfully assigned giantswarm/kvm-operator-685c8f7658-kdkkl to 10.0.4.108
  Normal  Pulling    38m   kubelet, 10.0.4.108  Pulling image "quay.io/giantswarm/kvm-operator:f5a6118557cd9b447d8d5906c5fdf7305849fdd6"
  Normal  Pulled     38m   kubelet, 10.0.4.108  Successfully pulled image "quay.io/giantswarm/kvm-operator:f5a6118557cd9b447d8d5906c5fdf7305849fdd6"
  Normal  Created    38m   kubelet, 10.0.4.108  Created container kvm-operator
  Normal  Started    38m   kubelet, 10.0.4.108  Started container kvm-operator
```

```
{⎈ |giantswarm-geckon:default} ➜  kvm-operator git:(vendor) k get pods -n 7ttg9               
NAME                            READY   STATUS    RESTARTS   AGE
master-34ous-747797ffb7-fvldd   4/4     Running   0          25m
worker-3ds4h-6fbd94d9fc-wjhvg   4/4     Running   0          25m
worker-9xcym-5cc44c6765-bktdj   4/4     Running   0          25m
worker-otk20-956dbbd7c-cqjxj    4/4     Running   0          25m
```

```
{⎈ |giantswarm-7ttg9:default} ➜  kvm-operator git:(vendor) k get nodes                                                                                                                    
NAME                            STATUS   ROLES    AGE   VERSION
master-34ous-747797ffb7-fvldd   Ready    master   14m   v1.15.5
worker-3ds4h-6fbd94d9fc-wjhvg   Ready    worker   14m   v1.15.5
worker-9xcym-5cc44c6765-bktdj   Ready    worker   14m   v1.15.5
worker-otk20-956dbbd7c-cqjxj    Ready    worker   14m   v1.15.5
```


```
{⎈ |giantswarm-7ttg9:default} ➜  kvm-operator git:(vendor) k get pods -A -w
NAMESPACE     NAME                                                   READY   STATUS    RESTARTS   AGE
giantswarm    chart-operator-7f96c79f6f-7d7ps                        1/1     Running   0          7m28s
giantswarm    tiller-deploy-5db95cf576-srr7t                         1/1     Running   0          10m
kube-system   calico-kube-controllers-6d4856d68f-jjqfn               1/1     Running   0          13m
kube-system   calico-node-9vksd                                      1/1     Running   0          13m
kube-system   calico-node-dxl76                                      1/1     Running   0          13m
kube-system   calico-node-hnb5z                                      1/1     Running   0          13m
kube-system   calico-node-qcf7n                                      1/1     Running   0          13m
kube-system   cert-exporter-4b6dv                                    1/1     Running   0          7m29s
kube-system   cert-exporter-dj76q                                    1/1     Running   0          7m29s
kube-system   cert-exporter-m988b                                    1/1     Running   0          7m22s
kube-system   cert-exporter-ntlbh                                    1/1     Running   0          7m30s
kube-system   coredns-7b76874c7b-fp89f                               1/1     Running   0          7m37s
kube-system   coredns-7b76874c7b-qhw89                               1/1     Running   0          7m38s
kube-system   k8s-api-healthz-master-34ous-747797ffb7-fvldd          1/1     Running   0          13m
kube-system   k8s-api-server-master-34ous-747797ffb7-fvldd           1/1     Running   0          13m
kube-system   k8s-controller-manager-master-34ous-747797ffb7-fvldd   1/1     Running   0          12m
kube-system   k8s-scheduler-master-34ous-747797ffb7-fvldd            1/1     Running   0          13m
kube-system   kube-proxy-4h8kx                                       1/1     Running   0          13m
kube-system   kube-proxy-7cgz6                                       1/1     Running   0          13m
kube-system   kube-proxy-qxgpp                                       1/1     Running   0          13m
kube-system   kube-proxy-qxs5c                                       1/1     Running   0          13m
kube-system   kube-state-metrics-586fbd9595-dftmj                    0/2     Pending   0          4m49s
kube-system   kube-state-metrics-6cddf57fb7-ldpxb                    2/2     Running   0          5m2s
kube-system   metrics-server-586d4684b4-5zr9j                        1/1     Running   0          10m
kube-system   net-exporter-6ljq5                                     1/1     Running   0          7m20s
kube-system   net-exporter-cjx94                                     1/1     Running   0          7m24s
kube-system   net-exporter-pxw2l                                     1/1     Running   0          7m27s
kube-system   net-exporter-shkk7                                     1/1     Running   0          7m25s
kube-system   nginx-ingress-controller-7c58c89bd4-7fnrv              1/1     Running   0          7m29s
kube-system   nginx-ingress-controller-7c58c89bd4-gc58d              1/1     Running   0          7m28s
kube-system   nginx-ingress-controller-7c58c89bd4-tvbtt              1/1     Running   0          6m48s
kube-system   node-exporter-4wtsm                                    1/1     Running   0          5m
kube-system   node-exporter-6vj42                                    1/1     Running   0          5m
kube-system   node-exporter-mc2nx                                    1/1     Running   0          5m
kube-system   node-exporter-rwbrf                                    1/1     Running   0          5m
```

**Comment by xh3b4sd** [comment]
Thank you Stevo. Can you tell a bit about your methodology on resolving this problem?

**Comment by xh3b4sd** [comment]
Merging as discussed.
